bootstrap-pull-request/src/prebuilt.ts

```diff
@@ -99,29 +91,80 @@ function assertIsPartialApplication(o: unknown): asserts o is PartialApplication
   if ('github.head-sha' in o.metadata.annotations) {
     assert(typeof o.metadata.annotations['github.head-sha'] === 'string', 'github.head-sha must be a string')
   }
+  assert('spec' in o, 'must have spec property')
+  assert(typeof o.spec === 'object', 'spec must be an object')
+  assert(o.spec !== null, 'spec must not be null')
+  assert('source' in o.spec, 'spec must have source property')
+  assert(typeof o.spec.source === 'object', 'source must be an object')
+  assert(o.spec.source !== null, 'source must not be null')
+  assert('path' in o.spec.source, 'source must have path property')
+  assert(typeof o.spec.source.path === 'string', 'path must be a string')
+}
+
+const parseApplicationManifest = async (applicationManifestPath: string): Promise<PartialApplication | Error> => {
+  let application
+  try {
+    application = yaml.load(await fs.readFile(applicationManifestPath, 'utf-8'))
+  } catch (error) {
+    if (error instanceof yaml.YAMLException) {
+      return error
+    }
+    throw error
+  }
+  try {
+    assertIsPartialApplication(application)
+  } catch (error) {
+    if (error instanceof assert.AssertionError) {
+      return error
+    }
+    throw error
+  }
+  return application
 }
 
 const writeServices = async (inputs: Inputs): Promise<void> => {
   const existingApplicationManifestPaths = await (
-    await glob.create(`${inputs.namespaceDirectory}/applications/**`, { matchDirectories: false })
+    await glob.create(`${inputs.namespaceDirectory}/applications/*.yaml`, { matchDirectories: false })
+  ).glob()
+  const prebuiltApplicationManifestPaths = await (
+    await glob.create(`${inputs.prebuiltDirectory}/applications/*.yaml`, { matchDirectories: false })
   ).glob()
+  for (const prebuiltApplicationManifestPath of prebuiltApplicationManifestPaths) {
+    const prebuiltApplication = await parseApplicationManifest(prebuiltApplicationManifestPath)
```

Comment:
We need to parse the prebuilt Application in order to extract the original `github.head-sha` value.